lib/ansible/galaxy/collection/__init__.py

```diff
@@ -594,41 +278,86 @@ def download_collections(collections, output_path, apis, validate_certs, no_deps
+    ):
+        for fqcn, concrete_coll_pin in dep_map.copy().items():  # FIXME: move into the provider
+            if concrete_coll_pin.is_virtual:
+                display.v(
+                    'Virtual collection {coll!s} is not downloadable'.
+                    format(coll=to_text(concrete_coll_pin)),
+                )
+                continue
 
-        requirements = []
-        with _display_progress(
-                "Starting collection download process to '{path!s}'".
-                format(path=output_path),
-        ):
-            for name, requirement in dep_map.items():
-                collection_filename = "%s-%s-%s.tar.gz" % (requirement.namespace, requirement.name,
-                                                           requirement.latest_version)
-                dest_path = os.path.join(output_path, collection_filename)
-                requirements.append({'name': collection_filename, 'version': requirement.latest_version})
-
-                display.display("Downloading collection '%s' to '%s'" % (name, dest_path))
-
-                if requirement.api is None and requirement.b_path and os.path.isfile(requirement.b_path):
-                    shutil.copy(requirement.b_path, to_bytes(dest_path, errors='surrogate_or_strict'))
-                elif requirement.api is None and requirement.b_path:
-                    temp_path = to_text(b_temp_path, errors='surrogate_or_string')
-                    temp_download_path = build_collection(requirement.b_path, temp_path, True)
-                    shutil.move(to_bytes(temp_download_path, errors='surrogate_or_strict'),
-                                to_bytes(dest_path, errors='surrogate_or_strict'))
-                else:
-                    b_temp_download_path = requirement.download(b_temp_path)
-                    shutil.move(b_temp_download_path, to_bytes(dest_path, errors='surrogate_or_strict'))
+            display.display(
+                "Downloading collection '{coll!s}' to '{path!s}'".
+                format(coll=concrete_coll_pin, path=to_native(b_output_path)),
+            )
 
-                display.display("%s (%s) was downloaded successfully" % (name, requirement.latest_version))
+            b_src_path = (
+                artifacts_manager.get_artifact_path
+                if concrete_coll_pin.is_concrete_artifact
+                else artifacts_manager.get_galaxy_artifact_path
+            )(concrete_coll_pin)
 
-            requirements_path = os.path.join(output_path, 'requirements.yml')
-            display.display("Writing requirements.yml file of downloaded collections to '%s'" % requirements_path)
-            with open(to_bytes(requirements_path, errors='surrogate_or_strict'), mode='wb') as req_fd:
-                req_fd.write(to_bytes(yaml.safe_dump({'collections': requirements}), errors='surrogate_or_strict'))
+            b_dest_path = os.path.join(
+                to_bytes(output_path, errors='surrogate_or_strict'),
```

Comment:
Can this use `b_output_path` from line 291?
```suggestion
                b_output_path,
```